src/lib.rs

```diff
@@ -178,10 +186,33 @@ impl Client {
         self.conn.flush().await?;
 
         match self.get_read_write_response().await? {
-            Response::Status(s) => Err(s.into()),
-            Response::Data(d) => d.ok_or(Status::NotFound.into()),
-            _ => Err(Status::Error(ErrorKind::Protocol(None)).into()),
+            Response::Status(s) => return Err(s.into()),
+            Response::Data(Some(items)) => {
+                for item in items {
+                    results.insert(item.key.clone(), Ok(Some(item)));
+                }
```

Comment:
This would result in a hashmap containing keys for which there is a cache hit and the corresponding Value payload received from the server, or a single error.  There is no way to map errors at a per-key level, as the server either will not respond in the case of a cache miss or will respond with a single error for the whole request command if there is a malformed key or something.  For this reason, it's not worth doing the work to transform the result set into a hashmap, and it's better to leave the function signature as-is.  By mapping to a hashmap, we are always forcing an iteration over the vector of results, effectively for no gain.  It's better to have the client do that work, and probably doesn't actually introduce any additional overhead at that layer since it's likely that the client would have to iterate over the list of keys that was passed in to the `get_multi` method in the first place.